src/execution.rs

```diff
@@ -320,16 +411,51 @@ impl Stanza {
                 shorthands: shorthands,
             };
             for statement in &self.statements {
-                statement
-                    .execute(&mut exec)
-                    .with_context(|| format!("Executing {}", statement))?;
+                statement.execute(&mut exec).or_else(|e| {
+                    Err(e).with_context(|| {
+                        let node = mat
+                            .captures
+                            .iter()
+                            .find(|c| c.index as usize == self.full_match_file_capture_index)
+                            .unwrap()
+                            .node;
+                        format!(
+                            "While executing statement {}\n{}\nin stanza\n{}\nmatching ({}) node\n{}",
+                            statement,
+                            Excerpt::from_source(tsg_path, tsg_source, statement.location()),
+                            Excerpt::from_source(tsg_path, tsg_source, self.location),
+                            node.kind(),
+                            Excerpt::from_source(
+                                source_path,
+                                source,
+                                Location::from(node.range().start_point)
+                            )
+                        )
+                    })
+                })?;
```

Comment:
I had initially toyed with adding a `Location` to each kind of error—or actually two, one for .tsg, one for target. Then I had each individual error site add an `Excerpt` to its description with whatever .tsg/source locations we could find.

I ended up adding them once and for all here, because it's _much_ simpler than adding locations and such to every kind of error, and unlike displaying excerpts at each error site, it's automatically exhaustive. However, it does come with the caveat that the precision of the error location is reduced relative to the error site information; e.g. for a duplicate scoped variable `node @expr.thing`, we get a caret pointing at `thing`; with the above approach we instead point at the start of the line. I think this was a worthwhile tradeoff, tho still potentially confusing in some cases.